dace/sdfg/sdfg.py

```diff
@@ -134,6 +134,13 @@ def free_symbols(self) -> Set[str]:
         result = set(
             map(str, dace.symbolic.symbols_in_ast(self.condition.code[0])))
         for assign in self.assignments.values():
+            # TODO: Proposal to skip free symbol extraction for assignments
+            # with dace.*. Currently failing in code generation
+            # if isinstance(assign, str) and assign.find('dace.') >= 0:
+            #     warnings.warn('Assignment expression {} cannot be evaluated '
+            #                   'for free symbols. Validation may miss '
+            #                   'undefined symbols.'.format(assign))
+            #     return set()
```

Comment:
could (should) be cleaned up

Reply:
This is describing an issue I cannot replicate now (maybe it was fixed in some other PR. Basically, scalar promotions where the scalar was set with a dace type (e.g., `a = dace.int32(3)`) failed. I will remove and we can make an issue if it appears again.